asn1tools/__init__.py

```diff
@@ -371,10 +377,11 @@ def _main():
         help='Hexstring to convert, or - to read hexstrings from standard input.')
     subparser.set_defaults(func=_do_convert)
 
-    # The 'shell' subparser.
-    subparser = subparsers.add_parser('shell',
-                                      description='An interactive shell.')
-    subparser.set_defaults(func=_do_shell)
+    if has_prompt_toolkit:
+        # The 'shell' subparser.
+        subparser = subparsers.add_parser('shell',
+                                          description='An interactive shell.')
+        subparser.set_defaults(func=_do_shell)
```

Comment:
Would probably be helpful to the user to always add the shell subparser, but fail with a nice error message if prompt_toolkit is not found.

Reply:
True, even better would be a fallback to built-in libraries like shelve (in place of diskcache) or cmd/readline (instead of prompt_toolkit), but that would be a lot of work and more changes. I left the subparser unchanged and added raise exception  in _do_shell instead.